expression/builtin_compare_test.go

```diff
@@ -283,23 +283,23 @@ func (s *testEvaluatorSuite) TestGreatestLeastFuncs(c *C) {
 		},
 		{
 			[]interface{}{"123a", "b", "c", 12},
-			float64(123), float64(0), false, false,
+			"c", "12", false, false,
 		},
 		{
 			[]interface{}{tm, "123"},
 			curTimeString, "123", false, false,
 		},
 		{
 			[]interface{}{tm, 123},
-			curTimeInt, int64(123), false, false,
+			curTimeString, "123", false, false,
 		},
 		{
 			[]interface{}{tm, "invalid_time_1", "invalid_time_2", tmWithFsp},
-			curTimeWithFspString, "invalid_time_1", false, false,
+			curTimeWithFspString, curTimeString, false, false,
```

Comment:
This is not compatible with mysql

Reply:
Ditto, comments about the compatibility are added.